### PR TITLE
feature: postgres as a new document db engine

### PR DESCRIPTION
**Pull request by ireznice**

### CODE REVIEW

File: web/module/requests.py [hunk 1]
```diff
@@ -22,20 +22,19 @@
...
-                    {
-                        'type': 'return_value',
-                        'response_id': '0',
-                        'is_last': True,
-                        'result': {
-                            'machine_id': req.machine,
-                            'state': req.state,
-                            'type': req.type
-                        }
+        asyncio.sleep(0.1)
```
Comment (petrSchreiber) [inline web/module/requests.py]: This is just transaction removed, right?

File: web/modeltr/connection.py [hunk 2]
```diff
@@ -42,21 +44,30 @@ def use(cls, alias=DEFAULT_CONNECTION_NAME):
...
-        return cls.__connections[alias]
+        connection = cls.__connections[alias]
+        try:
+            connection["connection"].client.reset()
+        except (psycopg2.InterfaceError, psycopg2.OperationalError, psycopg2.DatabaseError) as e:
+            connection["connection"] = cls(**connection["args"])
+        return cls.__connections[alias]["connection"]
 
 
+# TODO: delete
```
Comment (petrSchreiber) [inline web/modeltr/connection.py]: Maybe this TODO could be completed before merging?

File: web/modeltr/connection.py [hunk 3]
```diff
@@ -1,37 +1,39 @@
...
-        self.session.end_session()
-
-    # example usage: web.modeltr.connection.Connection(
-    #   host='localhost', authSource='test_database', replicaSet='rs0')
-    # auth source must be present here
     def __init__(self, **kwargs):
         self.__logger = logging.getLogger(__name__)
-        self.client = MongoClient(**kwargs)
-        self.database = kwargs['authSource']
+        for i in range(6):
```
Comment (petrSchreiber) [inline web/modeltr/connection.py]: This would be better to have in config file.

File: web/modeltr/connection.py [hunk 4]
```diff
@@ -1,37 +1,39 @@
-from pymongo import MongoClient
+# from pymongo import MongoClient
```
Comment (petrSchreiber) [inline web/modeltr/connection.py]: Line 1 could be removed.

File: vcenter/vcenter.py [hunk 5]
```diff
@@ -251,43 +335,53 @@ def start(self, uuid):
...
+                else:
+                    raise Exception('machine {} not found'.format(uuid))
+            except Exception:
+                settings.raven.captureException(exc_info=True)
+                self.__sleep_between_tries()
 
     def config_network(self, uuid, **kwargs):
         self.__logger.debug('config_network')
         self.__check_connection()
+        for i in range(6):
```
Comment (petrSchreiber) [inline vcenter/vcenter.py]: This would be nice to have in config.

File: vcenter/vcenter.py [hunk 6]
```diff
@@ -72,13 +77,27 @@ def search_for_snapshot(self, vm, snapshot_name):
         return None
 
     def __search_machine_by_name(self, vm_name):
-        objView = self.content.viewManager.CreateContainerView(self.content.rootFolder,
-                                                               [vim.VirtualMachine],
-                                                               True)
+        for cnt in range(5):
```
Comment (petrSchreiber) [inline vcenter/vcenter.py]: This would be better to have in config.

File: vcenter/vcenter.py [hunk 7]
```diff
@@ -46,11 +47,15 @@ def connect(self):
...
         self.content = si.content
         self._connected = True
+        self.vm_folders = VCenter.VmFolders(self)
 
     def idle(self):
         self.__check_connection()
         self.__logger.debug('keeping connection alive: {}'.format(self.content.about.vendor))
 
+    def __sleep_between_tries(self):
+        time.sleep(random.uniform(0, 3))
```
Comment (petrSchreiber) [inline vcenter/vcenter.py]: This range would be better in config.

File: vc-worker.py [hunk 8]
```diff
@@ -126,87 +162,90 @@ def action_others(conn, action, vc):
...
+
                 if action:
                     if mode == 'deploy':
                         action_deploy(conn, action, vc)
                     else:
                         action_others(conn, action, vc)
                 else:
                     idle_counter += 1
-                    if idle_counter > 20:
+                    if idle_counter > 60:
```
Comment (petrSchreiber) [inline vc-worker.py]: This would be better in config.

File: web/module/requests.py (hunk 1 above)
Comment (ireznice) [reply web/module/requests.py]: nope, must be here; a sync library is used in async environ.

File: web/modeltr/connection.py (hunk 2 above)
Comment (ireznice) [reply web/modeltr/connection.py]: right

File: web/modeltr/connection.py (hunk 3 above)
Comment (ireznice) [reply web/modeltr/connection.py]: done

File: web/modeltr/connection.py (hunk 4 above)
Comment (ireznice) [reply web/modeltr/connection.py]: yeah

File: vcenter/vcenter.py (hunk 5 above)
Comment (ireznice) [reply vcenter/vcenter.py]: right

File: vcenter/vcenter.py (hunk 6 above)
Comment (ireznice) [reply vcenter/vcenter.py]: right

File: vcenter/vcenter.py (hunk 7 above)
Comment (ireznice) [reply vcenter/vcenter.py]: yeah

**Pull request merged**